enumeratum-core/src/main/scala/enumeratum/Enum.scala

```diff
@@ -139,6 +146,27 @@ trait Enum[A <: EnumEntry] {
   def withNameLowercaseOnlyOption(name: String): Option[A] =
     lowerCaseNamesToValuesMap.get(name)
 
+  /**
+    * Returns an [[Right[A]]] for a given name, or a [[Left[String]]] if the name does not match any of the values'
+    * .entryName values, disregarding case.
+    */
+  def withNameInsensitiveEither(name: String): Either[String, A] =
```

Comment:
Can we get this updated to return `Left(NoSuchMember)` as well?

Reply:
Indeed, completely missed those ones !